### PR TITLE
add MintSwapCanonicalToken

### PR DESCRIPTION
**Pull request by xiaozhou**

### CODE REVIEW

**Review by i9 — APPROVED** [review]

File: contracts/pegged/tokens/MintSwapCanonicalToken.sol [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+pragma solidity >=0.8.9;
+
+import "./MultiBridgeToken.sol";
+
+/**
+ * @title Canonical token that supports multi-bridge minter and multi-token swap
+ */
+contract MintSwapCanonicalToken is MultiBridgeToken {
+    // each bridge token.balanceOf(this) tracks how much that bridge has already swapped
+    mapping(address => uint256) public tokenSwapCap; // each bridge token -> swap cap
```
Comment (i9) [inline contracts/pegged/tokens/MintSwapCanonicalToken.sol:12]: nit: term swapCap might cause confusion with swapBridgeForCanonical function, ie. someone might assume the cap is per swap call amount

File: contracts/pegged/tokens/MintSwapCanonicalToken.sol [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    event TokenSwapCapUpdated(address token, uint256 cap);
+
+    constructor(
+        string memory name_,
+        string memory symbol_,
+        uint8 decimals_
+    ) MultiBridgeToken(name_, symbol_, decimals_) {}
+
+    // msg.sender has bridge token and want to get canonical token
+    function swapBridgeForCanonical(address _bridgeToken, uint256 _amount) external {
```
Comment (i9) [inline contracts/pegged/tokens/MintSwapCanonicalToken.sol:23]: do we want to return canonical token amount?
it's not needed for this contract, but may be nice to have for future standard compatibility

File: contracts/pegged/tokens/MintSwapCanonicalToken.sol (hunk 1 above)
Comment (xiaozhou) [reply contracts/pegged/tokens/MintSwapCanonicalToken.sol]: changed to totalSwapCap

File: contracts/pegged/tokens/MintSwapCanonicalToken.sol (hunk 2 above)
Comment (xiaozhou) [reply contracts/pegged/tokens/MintSwapCanonicalToken.sol]: added

**Pull request merged**